### PR TITLE
Feature/624 Dialog creation refactoring using a service

### PR DESCRIPTION
**Pull request by kcinay055679**
In this branch, I refactored the way dialogs are opened by introducing a Dialog Service. This service simplifies the creation of a responsive dialog configuration, with responsiveness based on Bootstrap.

### CODE REVIEW

File: frontend/src/app/services/dialog.service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      ...config,
+    });
+  }
+
+  openConfirmDialog(translationKey: string) {
+    const tile = this.translationService.instant(`${translationKey}.TITLE`);
+    const text = this.translationService.instant(`${translationKey}.TEXT`);
+    return this.openConfirmDialogWithCustomText(tile, text);
+  }
+
```
Comment (peggimann) [inline frontend/src/app/services/dialog.service.ts:40]: add todo: // TODO: remove me

File: frontend/src/app/services/dialog.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  let service: DialogService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({ imports: [TranslateModule.forRoot()], providers: [TranslateService] });
+    service = TestBed.inject(DialogService);
+  });
+
+  it('should be created', () => {
+    expect(service).toBeTruthy();
+  });
```
Comment (peggimann) [inline frontend/src/app/services/dialog.service.spec.ts:18]: add some tests

File: frontend/src/app/components/objective/objective.component.ts [hunk 3]
```diff
@@ -96,22 +96,29 @@ export class ObjectiveComponent implements OnInit {
           action: 'todraft',
           dialog: {
             dialog: ConfirmDialogComponent,
-            data: { title: 'Objective', action: 'todraft' },
+            data: {
+              title: 'Check-in im Draft-Status',
```
Comment (peggimann) [inline frontend/src/app/components/objective/objective.component.ts:100]: add translation

File: frontend/src/app/shared/dialog/confirm-dialog/confirm-dialog.component.ts [hunk 4]
```diff
@@ -11,44 +12,13 @@ export class ConfirmDialogComponent implements OnInit {
...
-            error = 'DELETE_OBJECTIVE';
-            break;
-          case 'Key Result':
-            error = 'DELETE_KEY_RESULT';
-            break;
-        }
-        this.dialogText = this.translate.instant('INFORMATION.' + error);
-      }
-    }
+    this.dialogTitle = this.data.title || 'Are you sure?';
```
Comment (peggimann) [inline frontend/src/app/shared/dialog/confirm-dialog/confirm-dialog.component.ts:20]: update confirm logic and translations

**Review by peggimann — APPROVED** [review]

File: frontend/src/app/shared/dialog/confirm-dialog/confirm-dialog.component.ts (hunk 4 above)
Comment (kcinay055679) [reply frontend/src/app/shared/dialog/confirm-dialog/confirm-dialog.component.ts:20]: Will be fixed in pr #1072 
And behavior will be checked by [this](https://github.com/puzzle/okr/pull/1072/files#diff-f0558a6e2f00600d7f3a244322d44962271ef8d2369b1962e773f38c5e9ae642R31)

File: frontend/src/app/components/objective/objective.component.ts (hunk 3 above)
Comment (kcinay055679) [reply frontend/src/app/components/objective/objective.component.ts]: done :)

File: frontend/src/app/services/dialog.service.ts (hunk 1 above)
Comment (kcinay055679) [reply frontend/src/app/services/dialog.service.ts]: done

File: frontend/src/app/services/dialog.service.spec.ts (hunk 2 above)
Comment (kcinay055679) [reply frontend/src/app/services/dialog.service.spec.ts:18]: https://github.com/orgs/puzzle/projects/3/views/1?pane=issue&itemId=84808676&issue=puzzle%7Cokr%7C1088

**Review by peggimann — APPROVED** [review]
Nice! :100:

**Pull request merged**